libs/nft-contract/src/connect.ts

```diff
@@ -1,21 +1,30 @@
 import { Signer, Wallet } from "ethers";
 import type { TheCoinNFT } from "./types/TheCoinNFT";
 import { getContract } from "./contract";
+import { Network } from 'ethers/utils';
 
-export async function connectNFT(signer: Signer) : Promise<TheCoinNFT> {
-	const contract = await getContract();
-	if ((signer as Wallet).connect) {
+export function connectNFT(signer: Signer) : TheCoinNFT {
```

Comment:
By making this function sync, we can now use it during initialization

We'll probably want to do the same to TC contract.  Initialization is still painful